synapse_data_warehouse/synapse_aggregate/dynamic_tables/V2.49.0__unique_user_uploads_aggregate.sql

```diff
@@ -0,0 +1,93 @@
+USE SCHEMA {{database_name}}.synapse_aggregate; --noqa: JJ01,PRS,TMP
+
+CREATE OR REPLACE DYNAMIC TABLE UNIQUE_USER_UPLOADS
+    (
+	    GRANULARITY VARCHAR(16777216) COMMENT 'The dimension of the aggregate (e.g., YEARLY, MONTHLY, DAILY).',
+	    UNIQUE_USER_COUNT INT COMMENT 'The number of distinct unique users uploading during the aggregation period.',
+	    YEAR NUMBER(38,0) COMMENT 'Year of the aggregation period.',
+        MONTH NUMBER(38,0) COMMENT 'Month of the aggregation period.',
+        DAY NUMBER(38,0) COMMENT 'Day of the aggregation period.',
+        AGGREGATE_PERIOD_START DATE COMMENT 'The time when any change to the team was made (e.g. update of the team or a change to its members).',
+        AGGREGATE_PERIOD_STOP DATE COMMENT 'The time when any change to the team was made (e.g. update of the team or a change to its members).',
+	    SNAPSHOT_DATE DATE COMMENT 'Date the aggregation was calculated and stored in the table.',
+        IS_COMPLETE BOOLEAN COMMENT 'If true, then the aggregation period is complete.'
+    )
+    TARGET_LAG = '1 day'
+    WAREHOUSE = compute_xsmall
```

Comment:
Do we want to look into DYNAMIC table update strategy.  Right now i think this table gets re-written to, but really there's no need to re-compute aggregates that have already been done.  For example, 2024 aggregates only need to be computed once, we don't need to be running the aggregates for those.

Does dynamic tables do some sort of upsertting where it will only include rows that are unchanged (minus snapshot_date)

Reply:
I don't think the dynamic table is smart enough to avoid recomputing the aggregates. See the "optimizing aggregates" section in my design doc [here](https://sagebionetworks.jira.com/wiki/spaces/DPE/pages/3998548038/SNOW-187+Gold-layer+table+framework+aggregates#Example%3A-O3-KR2) for how we can cache an intermediate computation so that we can build the aggregate table faster. (This is overkill for now, imo).

Reply:
That's correct.

Because we're using `GROUP BY ROLLUP`, our aggregate dynamic tables can only be refreshed in full. Which means we need to re-compute all the aggregates each time.

In the future it would be nice to have agg tables that capture rates of change, so what did the YEARLY agg for 2024 look like in Q1 vs Q4? This is different from comparing Q1 vs. Q4 QUARTERLY aggs.